prow/cmd/pipeline/controller.go

```diff
@@ -665,8 +677,30 @@ const (
 	descMissingCondition = "missing end condition"
 )
 
-// prowJobStatus returns the desired state and description based on the pipeline status
-func prowJobStatus(ps pipelinev1alpha1.PipelineRunStatus) (prowjobv1.ProwJobState, string) {
+// prowJobStatus returns the worst state of any of the PipelineRuns passed to it
+func prowJobStatus(runs []*pipelinev1alpha1.PipelineRun) (prowjobv1.ProwJobState, string) {
+	var worstState *prowjobv1.ProwJobState
```

Comment:
Do you really need a pointer here since the returned value is not a pointer and probably the copy is cheap?